### ISSUE TITLE
[LSP Hover] No more syntax highlighting in hover floating window?

### ISSUE THREAD

**Issue by bluz71** [neovim/neovim#14134]
Hello Neovim LSP team,

Hover windows used to be syntax highlighted for me, but ever since commit [d49177af](https://github.com/neovim/neovim/commit/d49177afd9806f651c0177f9cc68c5a22dba2938) (*add custom syntax rules for lsp floating window*) I don't seem to getting syntax highlighting anymore. I did a Git Bisect to narrow down which commit changed hover highlighting.

Before commit `d49177af`, JavaScript `axios` hovering:

![axios_good](https://user-images.githubusercontent.com/11382509/111061341-fad4bf80-84f6-11eb-9721-abd1166d1884.png)

After, JavaScript `axios` hovering:

![axios_bad](https://user-images.githubusercontent.com/11382509/111061345-058f5480-84f7-11eb-882d-53f2b891692d.png)

Before commit `d49177af`, Dart Flutter hovering:

![flutter_good](https://user-images.githubusercontent.com/11382509/111061356-0fb15300-84f7-11eb-9b59-dcfcd884b3be.png)

After, Dart Flutter hovering:

![flutter_bad](https://user-images.githubusercontent.com/11382509/111061364-18098e00-84f7-11eb-911e-7f3f1d651f0c.png)

Hovering is initiated by this `K` mapping:

```lua
vim.api.nvim_buf_set_keymap(0, 'n', 'K', '<cmd>lua vim.lsp.buf.hover()<CR>', opts)
```

Am I doing something wrong?
I really like syntax highlighting in the hover window, I want to see it return.

Best regards.

**Comment by mjlbach** [comment]
Will be closed by: https://github.com/neovim/neovim/pull/14098 (there wasn't an associated issue before)

**Comment by bluz71** [comment]
Ok, I did try and find an issue before posting, it seems I am the first.

I look forward to the fix.

Cheers.

**Comment by mjlbach** [comment]
Oh yeah, just letting you know there was an outstanding PR :)

**Comment by mjlbach** [comment]
Closed by #14098, #14148

**Comment by bluz71** [comment]
Confirmed as fixed.

Many thanks @mjlbach.
